### DISCUSSION TITLE
ValueError Tensor on external RTX 2070

### DISCUSSION THREAD

**Discussion by ai-bits** [openai/whisper#1248] [Q&A]
Whisper runs nicely on my laptops: Razer w RTX 3080, Dell no Nvidia 32GB RAM even large-v2, albeit slow.

On a NUC 6th gen 32GB RAM w external RTX 2070 8GB VRAM with medium or small model same setup as above I get:
```
ValueError: Expected parameter logits (Tensor of shape (5, 51865)) of distribution Categorical(logits: torch.Size([5, 51865])) to satisfy the constraint IndependentConstraint(Real(), 1), but found invalid values:
tensor([[-43.3697,     -inf,     -inf,  ..., -56.2188, -57.4993, -72.2564],
        [     nan,...
```
and it doesn't detect English, but Korean and whatnot.
I know the 2070 doesn't like FP16, but -inf in the tensor doesn't seem to fit this argument and on the Razer with a PT-CPU-install it says it uses FP32 cuz the CPU can't FP16.
Thanks for any help
G.

**Comment by jongwook** [comment]
`-inf` should be fine for logits but I think the `nan`s are causing this error, as also observed in #1068. I'm not quite sure how exactly `nan` is happening. It could be related to subtle differences in certain PyTorch backends or the logit filtering steps. It'd be helpful tracking this down if there is a file that consistently causes this error..
